dds/DCPS/XTypes/DynamicData.cpp

```diff
@@ -2193,6 +2237,349 @@ const char* DynamicData::typekind_to_string(TypeKind tk) const
+    }
+    type_string += " = " + std::to_string(my_double) + "\n";
+    break;
+  }
+  case OpenDDS::XTypes::TK_FLOAT128: {
+    ACE_CDR::LongDouble my_longdouble;
+    if (dd.get_float128_value(my_longdouble, 0) != DDS::RETCODE_OK) {
+      ACE_ERROR_RETURN((LM_ERROR, "(%P|%t) Error: print_dynamic_data - failed to get_float128_value\n"), false);
+    }
+    type_string += " = " + std::to_string(my_longdouble) + "\n";
+    break;
+  }
+  case OpenDDS::XTypes::TK_BOOLEAN: {
+    ACE_CDR::Boolean my_bool;
+    if (dd.get_boolean_value(my_bool, 0) != DDS::RETCODE_OK) {
+      ACE_ERROR_RETURN((LM_ERROR, "(%P|%t) Error: print_dynamic_data - failed to get_boolean_value\n"), false);
+    }
+    type_string += " = " + std::to_string(my_bool) + "\n";
+    break;
+  }
+  case OpenDDS::XTypes::TK_BYTE: {
+    ACE_CDR::Octet my_byte;
+    if (dd.get_byte_value(my_byte, 0) != DDS::RETCODE_OK) {
+      ACE_ERROR_RETURN((LM_ERROR, "(%P|%t) Error: print_dynamic_data - failed to get_byte_value\n"), false);
+    }
+    type_string += " = " + std::to_string(my_byte) + "\n";
+    break;
+  }
+  case OpenDDS::XTypes::TK_CHAR8: {
+    ACE_CDR::Char my_char;
+    if (dd.get_char8_value(my_char, 0) != DDS::RETCODE_OK) {
+      ACE_ERROR_RETURN((LM_ERROR, "(%P|%t) Error: print_dynamic_data - failed to get_char8_value\n"), false);
+    }
+    type_string += OpenDDS::DCPS::String(" = \'") + my_char + "\'\n";
+    break;
+  }
+  case OpenDDS::XTypes::TK_CHAR16: {
+    ACE_CDR::WChar my_wchar;
+    if (dd.get_char16_value(my_wchar, 0) != DDS::RETCODE_OK) {
+      ACE_ERROR_RETURN((LM_ERROR, "(%P|%t) Error: print_dynamic_data - failed to get_char16_value\n"), false);
+    }
+    type_string += " = \'" + std::to_string(my_wchar) + "\'\n";
+    break;
+  }
+  case OpenDDS::XTypes::TK_STRING8: {
+    ACE_CDR::Char* my_string = 0;
+    if (dd.get_string_value(my_string, 0) != DDS::RETCODE_OK) {
+      ACE_ERROR_RETURN((LM_ERROR, "(%P|%t) Error: print_dynamic_data - failed to get_string_value\n"), false);
+    }
+    type_string += OpenDDS::DCPS::String(" = \"") + my_string + "\"\n";
+    break;
+  }
+  case OpenDDS::XTypes::TK_STRING16: {
+    ACE_CDR::WChar* my_wstring = 0;
+    if (dd.get_wstring_value(my_wstring, 0) != DDS::RETCODE_OK) {
+      ACE_ERROR_RETURN((LM_ERROR, "(%P|%t) Error: print_dynamic_data - failed to get_wstring_value\n"), false);
+    }
+    //type_string += " = \"" + my_wstring + "\"\n"; TODO CLAYTON: Find how to cast WChar* to a string
+    break;
+  }
+  // case OpenDDS::XTypes::TK_BITMASK:
+  // case OpenDDS::XTypes::TK_BITSET:
+  case OpenDDS::XTypes::TK_ALIAS: {
+    if (!print_dynamic_data(dd, dt->get_descriptor().base_type, type_string, indent)) {
+      ACE_ERROR((LM_ERROR, "(%P|%t) print_dynamic_data: failed to read alias\n"));
+    }
+    break;
+  }
+  case OpenDDS::XTypes::TK_SEQUENCE: {
+    OpenDDS::DCPS::String temp_indent = indent;
+    indent += "  ";
```

Comment:
The `Printable` object can manage indents. It also allows what the indent is to be changed.

Reply:
Is this change necessary? It seems like unnecessary complexity for this case.

Reply:
This is exactly what the `Printer` was made for is all I'm trying to say. We don't have to do anything about it now, but when we rewrite this to use `ValueReader`/`ValueWriter` we should do it make by making a `PrinterValueWriter`.